### DISCUSSION TITLE
some question about use zephir develop php extension ？

### DISCUSSION THREAD

**Discussion by 010test** [zephir-lang/zephir#2368] [Q&A]
hello erveryone, i‘m a beginner, i taste use zephir to write some commonly used php function as php extension, i  read the development documentation and finished, then i started to write, although success, but i still has some question like follows：

for this php function, the problem is in the code

```
    /**
     * Get the start timestamp and end timestamp of the month in which a timestamp is located
     * @param int $timestamp
     * @return array
     */
    final public static function getMonthBeginEnd(int timestamp = 0) -> array {
        array arr = [];

        if !timestamp {
            //here, why must add (int)? otherwise complie error, the function time() return value is originally an int
            let timestamp = (int)time();
        }

        var year = date("Y", timestamp), month = date("m", timestamp), d = date("t", strtotime(year . "-" . month));

        let arr["begin"] = strtotime(year . "-" . month);
        let arr["end"] = mktime(23, 59, 59, month, d, year);

        return arr;
        //the return why not can write like follow?
        return ["begin" => strtotime(year . "-" . month), "end" => mktime(23, 59, 59, month, d, year)];
    }
```

how to write like the follow code ?

```
for var i = date("d", startTime); i <= date("d", endTime); let i++ {......} 
```

can't find example in the documentation, now i write it replace while. 

my english is not very good, the question it should be clear, thank god for answer !

**Comment by Jeckerson** [comment]
You will need to declare `date("d", endTime)` into separate variable.

**Reply by 010test** [reply]
thanks, Please help me with the other two questions, thanks a lot.

**Reply by danhunsaker** [reply]
The reason the `return` can't be written as you propose is the same as the reason you couldn't write the `for` loop the way you proposed. You have to create variables before you can use them - raw values can only be returned if they are basic "scalar" types. For complex types, such as arrays, you need to have a variable to store the result in so that there's something to pass back to the caller. C passes complex types by reference (well, sort of; pointers are a bit more complex than that), meaning you have to have a variable from which to return the reference (or, well, memory address). Since Zephir compiles to C, it has to be quite a lot stricter about that sort of thing than pure PHP, which abstracts all that away by making _every_ value a complex type.

I could go on for hours, but the short answer is: You have to have a variable to shove complex operations into.

It's a bit more verbose, but it's also quite a lot safer, so it works out.

As to needing to cast `time()` to `int` ... I'd need to do some digging to say for sure.
